### PR TITLE
Cloning GUI

### PR DESCRIPTION
**Pull request by Aranclanos**
### Purpose
Adds an interface for the cloning computer

### Open Questions and Pre-Merge TODOs

- [X]  This fix is tested on the same branch it is PR'ed to.
- [X]  I correctly commented my code
- [X]  My code is indented with tabs and not spaces
- [X]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [X]  This PR does not bring up any new compile errors
- [X]  This PR has been tested in editor
- [X]  This PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)

### CODE REVIEW

**Comment by fomalsd** [comment]
got a gif?

**Comment by Bod9001** [comment]
Can it clone people or is it just the interface for the cloner?

**Comment by Aranclanos** [comment]
Interface, but it handles record deletion and saving

**Comment by Bod9001** [comment]
ok, all cool

**Comment by Bod9001** [comment]
https://cdn.discordapp.com/attachments/312454684021620736/582922526171201566/G98yiuhkj87tyuihjkIF.gif
a gif

File: UnityProject/Assets/Scripts/UI/Cloning/GUI_Cloning.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        recordList.Clear();
+        recordList.AddItems(cloningRecords.Count);
+
+        for (int i = 0; i < cloningRecords.Count; i++)
+        {
+            GUI_CloningRecordItem item = recordList.Entries[i] as GUI_CloningRecordItem;
+			item.gui_Cloning = this;
+            item.cloningRecord = cloningRecords[i];
+			item.SetValues();
+            item.gameObject.SetActive(true);
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/UI/Cloning/GUI_Cloning.cs:122]: Don't call SetActive in tabs yourself, it messes up elements.

File: UnityProject/Assets/Scripts/UI/Cloning/GUI_CloningRecordItem.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public CloningRecord cloningRecord;
+	public GUI_Cloning gui_Cloning;
+
+    public NetLabel recordName;
+    public NetLabel recrodScanID;
+
+    public void SetValues()
+    {
+        recordName.Value = cloningRecord.Name;
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/UI/Cloning/GUI_CloningRecordItem.cs:16]: Use
```cs
recordName.SetValue = cloningRecord.Name;
recrodScanID.SetValue = "Scan ID " + cloningRecord.ScanID;
```
To send updates instantly

File: UnityProject/Assets/Scripts/UI/Cloning/GUI_CloningItemList.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GUI_CloningItemList : NetUIDynamicList
+{
+    public void AddItems(int count)
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/UI/Cloning/GUI_CloningItemList.cs:7]: This class is indented by spaces, please reformat to tabs

File: UnityProject/Assets/Scripts/UI/Cloning/GUI_Cloning.cs [hunk 4]
```diff
@@ -0,0 +1,126 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class GUI_Cloning : NetTab
+{
+	public CloningConsole CloningConsole;
+    public GUI_CloningItemList recordList = null;
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/UI/Cloning/GUI_Cloning.cs:9]: This class is partially indented by spaces, please reformat to tabs

File: UnityProject/Assets/Scripts/UI/Cloning/GUI_CloningRecordItem.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class GUI_CloningRecordItem : DynamicEntry
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/UI/Cloning/GUI_CloningRecordItem.cs:6]: This class is partially indented by spaces, please reformat to tabs

**Review by fomalsd — CHANGES_REQUESTED** [review]
I haven't checked it out ingame yet, but I've seen the gif:
could you please fix records count on the main page?
And also see notes below

**Comment by Aranclanos** [comment]
Just realised I pushed to unitystation instead to my fork, I made a new PR with the new changes

**Pull request closed without merging**